src/microbe_stage/MembraneWaterRipple.cs

```diff
@@ -0,0 +1,572 @@
+            screenPos.Y > -thresholdY && screenPos.Y < thresholdY;
+    }
+
+    /// <summary>
+    ///   Updates the camera position cache for distance calculations
+    /// </summary>
+    private void UpdateCameraCache()
+    {
+        currentCamera = GetViewport()?.GetCamera3D();
+        if (currentCamera == null)
+        {
+            isCameraPositionValid = false;
+            return;
+        }
+
+        Vector3 cameraPos = currentCamera.GlobalPosition;
+
+        // Only update if camera moved significantly
+        if (!isCameraPositionValid || lastCameraPosition.DistanceSquaredTo(cameraPos) > 0.25f)
+        {
+            lastCameraPosition = cameraPos;
+
+            if (parentMembrane != null)
+            {
+                lastCameraDistance = parentMembrane.GlobalPosition.DistanceTo(cameraPos);
+            }
+
+            isCameraPositionValid = true;
+            UpdateLodLevel();
+        }
+    }
+
+    /// <summary>
+    ///   Updates the LOD level based on camera distance
+    /// </summary>
+    private void UpdateLodLevel()
+    {
+        if (!isCameraPositionValid || parentMembrane == null)
+            return;
+
+        LodLevel newLodLevel;
+        float membraneRadius = GetMembraneRadius();
+
+        // Scale the distance thresholds based on the membrane size
+        float sizeScale = Math.Clamp(membraneRadius / 5.0f, 1.0f, 2.0f);
+
+        if (lastCameraDistance > VERY_LOW_LOD_DISTANCE * sizeScale)
+        {
+            newLodLevel = LodLevel.VeryLow;
+        }
+        else if (lastCameraDistance > LOW_LOD_DISTANCE * sizeScale)
+        {
+            newLodLevel = LodLevel.Low;
+        }
+        else if (lastCameraDistance > HIGH_LOD_DISTANCE * sizeScale)
+        {
+            newLodLevel = LodLevel.Medium;
+        }
+        else
+        {
+            newLodLevel = LodLevel.High;
+        }
+
+        // If LOD level changed update subdivision
+        if (newLodLevel != currentLodLevel)
+        {
+            currentLodLevel = newLodLevel;
+            UpdateMeshSubdivision();
+        }
+    }
+
+    /// <summary>
+    ///   Updates mesh subdivision based on current LOD level
+    /// </summary>
+    private void UpdateMeshSubdivision()
+    {
+        if (planeMesh == null)
+            return;
+
+        int subdivision;
+
+        switch (currentLodLevel)
+        {
+            case LodLevel.VeryLow:
+                subdivision = VERY_LOW_LOD_SUBDIVISION;
+                break;
+
+            case LodLevel.Low:
+                subdivision = LOW_LOD_SUBDIVISION;
+                break;
+
+            case LodLevel.Medium:
+                subdivision = MEDIUM_LOD_SUBDIVISION;
+                break;
+
+            case LodLevel.High:
+                subdivision = HIGH_LOD_SUBDIVISION;
+                break;
+
+            default:
+                subdivision = MEDIUM_LOD_SUBDIVISION;
+                break;
+        }
+
+        planeMesh.SubdivideWidth = subdivision;
```

Comment:
How much performance does this take? I imagine this regenerates the mesh which I can't imagine is good for performance. Probably better to have just static instances of each plane mesh and swap between them dynamically.